rqt_bag/src/rqt_bag/timeline_menu.py

```diff
@@ -45,22 +45,18 @@ def __init__(self, popup_name, timeline, viewer_type, topic):
         self._viewer_type = viewer_type
         self._topic = topic
         self._viewer = None
+        self._is_listening = False
+
 
     def hideEvent(self, event):
-        # check that we were actually hidden. hide events can also happen if a
-        # window is minimized, but isVisible will still be true if a window is
-        # only minimized and not closed
-        if not self.isVisible():
-            # remove the viewer
-            if self._viewer:
-                self._timeline.remove_listener(self._topic, self._viewer)
-                self._viewer.close()
-                self._viewer = None
+        if self._is_listening:
+            self._timeline.remove_listener(self._topic, self._viewer)
+            self._is_listening = False
 
-            # clean out the layout
-            while self.layout().count() > 0:
-                item = self.layout().itemAt(0)
-                self.layout().removeItem(item)
+    def showEvent(self, event):
```

Comment:
While the parent method is indeed empty for a QWidget (https://qt.gitorious.org/qt/qt/source/3f2f403191ea5a38c66b79840e3a33e05827d580:src/gui/kernel/qwidget.cpp#L9572) that:
* might change in the future and
* is not the case for other widgets, e.g. combo box (https://qt.gitorious.org/qt/qt/source/3f2f403191ea5a38c66b79840e3a33e05827d580:src/gui/widgets/qcombobox.cpp#L2836)

Therefore I would highly suggest to always call the parent method when overriding it.